linode/manager
# upcoming: [M3-7659] - Add `user_type` to `/profile` endpoint

## Description 📝
We're updating our `/profile` endpoint to return the user role associated with Parent/Child Accounts by adding a new `user_type` field. Previous we had to get this information from the `/users` or `/users/:user` endpoint. This will allow us to cut down on the amount of queries we need in order to determine the user role.

## Changes  🔄
- Add `user_type` to `type Profile` in APIv4
- You'll see with these changes we were able to remove `useAccountUser` query hooks
- For MSW, you'll have to manually change the `user_type` for `*/profile` similar to how we do for `*/account/users/:user`. 

## How to test 🧪

### Prerequisites
- One account with full access
- One account with restricted access

### Reproduction steps
- Go to http://localhost:3000/account/billing

### Verification steps 
1. Log into any restricted account with parent/child feature flag **disabled**, observe the `Edit` and `Add Payment Method` buttons are disabled.
2. Turn parent/child feature flag on with Mock Service Workers
    - Observe the `Edit` and `Add Payment Method` buttons are enabled (by default we're in a `parent` state)
    - Then change `parent` to `child` [on this line](https://github.com/linode/manager/blob/9ad5df4411d3c21a2d78645b2e73e29f105aed5d/packages/manager/src/mocks/serverHandlers.ts#L554)
    - Observe the `Edit` and `Add Payment Method` buttons are disabled.
    - Now change MSW from `child` to `proxy` and observe the `Edit` and `Add Payment Method` buttons are enabled.

## As an Author I have considered 🤔

*Check all that apply*
